src/controller.ts

```diff
@@ -0,0 +1,147 @@
+import config from 'config';
+import Docker from 'dockerode';
+import fsAsync from 'node:fs/promises';
+import path from 'node:path';
+import { clearIntervalAsync } from 'set-interval-async';
+import { setIntervalAsync } from 'set-interval-async/dynamic';
+import type { SetIntervalAsyncTimer } from 'set-interval-async';
```

Comment:
We use now ESM modules, which means that the import syntax is different.

Instead of `const toto = require('toto');` we use `import toto from 'toto';`. There is a lot of drama around this change in the NodeJS community, but the new version is supposedly better. I'm not convinced though.

You can also note that we use node modules prefixed by `node:` and the new `node:fs/promises` API which is nice.